nextcloud-startup-script.sh

```diff
@@ -330,28 +330,58 @@ bash $SCRIPTS/nextcloud_configuration.sh
 bash $SCRIPTS/additional_apps.sh
 
 clear
-# Change passwords
+### Change passwords
 # CLI USER
-print_text_in_color "$ICyan" "For better security, change the system user password for [$(getent group sudo | cut -d: -f4 | cut -d, -f1)]"
-any_key "Press any key to change password for system user..."
+msg_box "For better security, we will now change the password for the UNIX-user (Ubuntu)"
+UNIXUSER="$(getent group sudo | cut -d: -f4 | cut -d, -f1)"
 while :
 do
-    sudo passwd "$(getent group sudo | cut -d: -f4 | cut -d, -f1)" && break
+    UNIX_PASSWORD=$(input_box_flow "Please type in the new password for the UNIX-user [$UNIXUSER]")
+    if [[ "$UNIX_PASSWORD" == *" "* ]]
+    then
+        msg_box "Please don't use spaces"
+    else
+        break
+    fi
 done
-echo
-clear
+check_command echo "$UNIXUSER:$UNIX_PASSWORD" | sudo chpasswd
+unset UNIX_PASSWORD
+
 # NEXTCLOUD USER
 NCADMIN=$(occ_command user:list | awk '{print $3}')
-print_text_in_color "$ICyan" "The current admin user in Nextcloud GUI is [$NCADMIN]"
-print_text_in_color "$ICyan" "We will now replace this user with your own."
-any_key "Press any key to replace the current (local) admin user for Nextcloud..."
-# Create new user
+msg_box "We will now change the username and password for the Nextcloud admin user"
 while :
 do
-    print_text_in_color "$ICyan" "Please enter the username for your new user:"
-    read -r NEWUSER
-    sudo -u www-data $NCPATH/occ user:add "$NEWUSER" -g admin && break
+    NEWUSER=$(input_box_flow "Please type in the name of the Nextcloud admin user. It must differ from [$NCADMIN].\nThe only allowed character are: 'a-z', 'A-Z', '0-9', and '_.@-'")
+    if [[ "$NEWUSER" == *" "* ]]
+    then
+        msg_box "Please don't use spaces."
+    elif [ "$NEWUSER" = "$NCADMIN" ]
+    then
+        msg_box "This username is already in use. Please choose a different one."
+    elif [[ "$NEWUSER" =~ [^a-zA-Z0-9_.@-] ]]
```

Comment:
Why is the '^' needed?

Reply:
Don't know actually, but it works, so I'm satisfied. :)

Got inspiration from this: https://www.golinuxcloud.com/check-if-string-contains-numbers-shell-script/

Reply:
Okay cool, but it has to be outside of the `[]` otherwise it will be an allowed character.

Reply:
Try for yourself. :)

Reply:
I guess it has to be `elif [[ "$NEWUSER" =~ ^[a-zA-Z0-9_.@-]+$ ]]`

Reply:
better is most likely `elif [[ "$NEWUSER" =~ ^[a-zA-Z][a-zA-Z0-9_.@-]+$ ]]`
(we can only allow a letter at the beginning this way)

Reply:
sry I meant `elif ! [[ "$NEWUSER" =~ ^[a-zA-Z][a-zA-Z0-9_.@-]+$ ]]`

Reply:
Can you please test all possible combinations before we commit this?

Reply:
The current implementation worked in my testing.

Reply:
I made a little test function:

```
while :
do
read -r NEWUSER
    #if [[ "$NEWUSER" =~ [^a-zA-Z0-9_.@-] ]]
    if ! [[ "$NEWUSER" =~ ^[a-zA-Z0-9_.@-]+$ ]]
    then
        echo "Allowed characters are: a-z', 'A-Z', '0-9', and '_.@-'\nPlease try again."
    else
        echo "VALID PASSWORD!"
    fi
done
```

Please use that and test different scenarios. We still don't cover ÅÄÖ for example, it shows up as valid, and if that happens, the whole thing never stops looping.

Reply:
Both yourrs and mine produce exactly the same result btw.

Reply:
okay great, then just use the one that you like more :D

Reply:
I trust you if you say it works :)
I am currently working on the new smbserver-script ;)

Reply:
We still need to figure out how to not allow only english characters, *not* ÅÄÖ and the german U, double S, and so on.

Regarding SMB-Server, that's another project. But maybe you can convince me, we'll see. :)